assets/js/modules/adsense/util/report-date-range-args.test.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Tests for reporting date range utilities.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { generateDateRangeArgs } from './report-date-range-args';
+
+describe( 'AdSense report date range: generateDateRangeArgs', () => {
+	it( 'should throw if a `dates` object is not provided', () => {
+		expect( () => generateDateRangeArgs() ).toThrow( 'dates object is required' );
+	} );
+
+	it( 'should throw if `startDate` argument is not provided', () => {
+		expect( () => generateDateRangeArgs( { endDate: '2021-01-27' } ) ).toThrow( 'startDate is required' );
+	} );
+
+	it( 'should throw if `endDate` argument is not provided', () => {
+		expect( () => generateDateRangeArgs( { startDate: '2021-01-01' } ) ).toThrow( 'endDate is required' );
+	} );
+
+	it( 'should return an object containing a `d` key, the value of which is the `startDate` argument with "-" replaced with "/", concatenated to the `endDate` argument also with "-" replaced with "/", using `-` to join the two strings', () => {
```

Comment:
This reads a bit overly verbose and is a bit too much detail regarding the implementation. While very explicit, it likely isn't all necessary in the output, even if this test fails.

It's okay for some of the logic to be conveyed by the test itself, as long as it is clear. In this case, I think it's quite clear what it's doing so a more concise name would be easier to read without hurting the clarity of the intent.

Reply:
LOL yes, now I read it again it does seem a bit long